common4j/src/main/com/microsoft/identity/common/java/commands/SilentTokenCommand.java

```diff
@@ -109,12 +108,24 @@ public AcquireTokenResult execute() throws Exception {
                 } catch (UiRequiredException | ClientException e) {
                     if (e.getErrorCode().equals(OAuth2ErrorCode.INVALID_GRANT) // was invalid_grant
                             && controllers.size() > ii + 1) { // isn't the last controller we can try
+                        exceptionFromFirstController = exceptionFromFirstController == null ? e : exceptionFromFirstController;
```

Comment:
It seems the logic here is such that "_If not already set then this must be first controller exception else this was not first_" but I am thinking that code might be easier to understand if you just do the following:

```java
if (ii == 0) {
   exceptionFromFirstController = e;
}
```